resume-buddy/src/main/java/com/google/sps/Match.java

```diff
@@ -26,31 +53,116 @@ private Match() {}
   }
 
   /**
-   * FCFS algorithm to match reviewers with reviewees. When two people are matched, their entities
-   * are deleted ('Reviewee' and 'Reviewer') from datastore and new entity ('Match') for their pair
-   * is created
+   * Algorithm to match reviewers with reviewees based on their schools, industry, experience, and
+   * degree. When two people are matched, their entities are deleted ('Reviewee' and 'Reviewer')
+   * from datastore and new entity ('Match') for their pair is created
    */
   public static void match(List<Entity> reviewees, List<Entity> reviewers) {
-    // TODO: Update algorithm based on criteria
+    // List that hold matchpoint with the possible pair of matched people
+    List<Pair<Integer, Pair<Entity, Entity>>> rankedMatches =
+        new ArrayList<Pair<Integer, Pair<Entity, Entity>>>();
 
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
-    int numberOfMatched = Math.min(reviewees.size(), reviewers.size());
 
-    for (int i = 0; i < numberOfMatched; i++) {
-      Entity reviewer = reviewers.get(i);
-      Entity reviewee = reviewees.get(i);
+    // For every reviewee-reviewer pair, count their match point
+    for (Entity reviewee : reviewees) {
+      for (Entity reviewer : reviewers) {
+        // TODO: Try different pointing systems to see which works the best
+        int matchPoint = 0;
 
-      Entity matchEntity = new Entity("Match");
+        String revieweeCareer = (String) reviewee.getProperty("career");
+        String reviewerCareer = (String) reviewer.getProperty("career");
+        if (revieweeCareer.equals(reviewerCareer)) {
+          matchPoint += 3;
```

Comment:
The points system is a great idea!

You can make these points constants at the top of this class. Eg.
```
private static final int CAREER_POINTS = 3;
private static final int SCHOOL_POINTS = 2;
...
```